### DISCUSSION TITLE
[2023-05-02] GUI Demo Day

### DISCUSSION THREAD

**Discussion by farmaazon** [enso-org/enso#6508] [Weekly Demo Day]
I will cover both me and @Frizi and @wdanilo to some extent.

Shortly, we finally managed to make working list editor in nodes!

https://user-images.githubusercontent.com/3919101/235682274-0655f392-3ae0-46e5-ab06-2007cce2085d.mp4

You can test it on develop; the feature flag is no longer required, as it's enabled by default. Adding, removing and dragging should work. You can even drag elements between different nodes. The nested list are not supported, though (they are displayed, but not as the list editor widget).

It was a tough fight. In the process, @Frizi created a debug mode there every display object is displayed as DIV element with its position and size, so you can check them in the browser dev tools. To enable the mode, change the `ENABLE_DOM_DEBUG` constant. We also fixed some bugs in EnsoGL list editor component.

![image](https://user-images.githubusercontent.com/3919101/235683662-8168232d-b93c-4058-abca-b72dde727588.png)

**Comment by Procrat** [comment]
1. #6430
2. #6421

Before:

![Screenshot from 2023-04-04 16-55-39](https://user-images.githubusercontent.com/292128/229836099-e38ee6d1-e659-416c-81d0-18a50fcafd71.png)
After:

https://user-images.githubusercontent.com/607786/234297411-8c0b3851-5977-4ca5-b3b4-5b0782510e14.mp4

3. #6474: Adds a button to return to the project dashboard. This part just fires a DOM event (see #6399).

https://user-images.githubusercontent.com/607786/235687669-ab04339f-0f07-439a-9cd3-59d96815edaa.mp4

**Comment by MichaelMauderer** [comment]
1.  Fixed "Can not click on a dropdown to change execution mode if the mouse pointer is above the text #6371 " in #6487 by fixing the  layer ordering between dropdown and breadcrumbs backgrounds.
2.  Fixed "Node selection issues #5628" with #5628. Uses the bounding box of nodes without the visualization for determining area selection. 

https://user-images.githubusercontent.com/1428930/235451552-d99cf4c7-4420-4412-b553-e9bf6615a8a8.mp4

3. Finalized Execution Context chooser (#6434) 
4. Merged Play Button controller integration in (#6427)

https://user-images.githubusercontent.com/1428930/234429549-2e25a8e2-a677-475c-a4b2-49f00421b9ad.mp4

**Comment by wdanilo** [comment]
I was working on several things:
1. Fixing bugs in List Editor to support the use case @farmaazon and @Frizi were working on.
2. Extending the List Editor to support multiple list editors, icon when the items are removed, adding possibility to drag items between list editors, adding new way to the mouse cursor to manage dragged objects in a general way (see video below).
3. Investigated a few bugs, including why the fullscreen visualizations scrolling moves nodes under them.
4. Refactored lider implementation to work more generic and correctly with our new layouts.
5. Started working on an experimental performance fixes (don't know yet if they are the best way to approach our problems, but the results are very promising). More info soon.


https://user-images.githubusercontent.com/1623053/235690544-515299f6-96fd-4241-8574-621b2c731d56.mov

**Comment by hubertp** [comment]
Engine:
- Improve inlining of <| on (GraalVM EE)   https://github.com/enso-org/enso/pull/6384
- And an even more generic solutiion to the above -  Introducing @BuiltinMethod.inlineable and InlineableNode https://github.com/enso-org/enso/pull/6442
- Reporting only unique warnings (reported last week but greatly improved since) https://github.com/enso-org/enso/pull/6372
- Resolve initialization problems on Windows with vcs/init due to long hostname resolution calls https://github.com/enso-org/enso/pull/6463

**Comment by vitvakatu** [comment]
First of all, last week, we finally finished [Execution contexts in GUI epic](https://github.com/enso-org/enso/issues/5899). Thank you @MichaelMauderer and @Procrat, for your hard work.

(thanks @jdunkerley for the videos)

https://user-images.githubusercontent.com/6566674/235691370-10964d2a-b833-421d-bd9a-e7613c57b3ef.mp4

https://user-images.githubusercontent.com/6566674/235691426-a3f2d65b-9fc3-4fd0-8b9a-d6a103a48ef8.mp4

I fixed the bug with event handling accidentally introduced by me: https://github.com/enso-org/enso/pull/6396
Also, a quick fix for project name focus: https://github.com/enso-org/enso/pull/6432

With help from @Frizi, I fixed the complex bug with named arguments becoming doubled: https://github.com/enso-org/enso/pull/6422.

Finally, I was doing a bunch of QA:
1. https://github.com/enso-org/enso/pull/6341
2. https://github.com/enso-org/enso/pull/6365
3. https://github.com/enso-org/enso/pull/6366
4. https://github.com/enso-org/enso/pull/6421
5. https://github.com/enso-org/enso/pull/6321

**Comment by kazcw** [comment]
- Fixed disappearing icons (#6125), caused by a FRP-concurrency bug; fixed several GL API bugs found while debugging (use of dangling Textures; leaking Framebuffers; incorrect MAG/MIN filter parameters).
- Fixed disappearing text (#6477); it appears we had never supported multiple fonts in the same layer.

**Comment by mwu-tow** [comment]
1. Fixing issue with URL handling on macOS https://github.com/enso-org/enso/issues/6362.
2. Fixed issue where IDE spawning Project Manager caused a console window appearing on Windows https://github.com/enso-org/enso/issues/6360.
3. Tons of debugging for https://github.com/enso-org/enso/issues/6447.
4. IDE logging https://github.com/enso-org/enso/pull/6478.
